pkg/util/url_test.go

```diff
@@ -8,28 +8,63 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
-func TestUrl(t *testing.T) {
+func TestJoinURLFragments(t *testing.T) {
+	t.Parallel()
```

Comment:
What's the exact benefit of calling `t.Parallel`? I haven't seen this in action before, so could use some insight into why it's applied here.

Reply:
@aknuds1 It tells the Go test runner that this test is safe to run in parallel with other tests also marked with `t.Parallel`. In theory, if we did this for all tests which are truly units (i.e. testing pure functions or contained environments) we should be able to run the tests several times faster.

Reply:
@sakjur Interesting. AFAICT sub-tests are also run in parallel (in their own goroutines). Will Go run all (non-sub-)tests in the test suite in serial, unless you use `t.Parallel`?

Reply:
Yes! I chose not to do that optimization here since these validations are nearly instantaneous anyways so running the top in parallel *should* net similar benefits with less overhead, but they are serial by default.

Reply:
That is to say sub-tests run serially by default, but in their own Goroutines. There's a great post about this in the Go blog, I'll try to dig it up

Reply:
https://blog.golang.org/subtests

I think my subtests should be made parallel _as well_ for this to really work 🤔

Reply:
Thanks @sakjur - good reading. So sub-tests will run in serial in relation to each other, even if they are in separate goroutines? I'm slightly confused.